tests/spec/examples.py

```diff
@@ -795,6 +797,16 @@ def it_runs_before_starting_a_service(self):
             norm=norm.pgctl,
         )
 
+    @pytest.mark.usefixtures('in_example_dir')
+    def it_runs_before_debugging_a_service(self):
+        proc = Popen(('setsid', 'pgctl', 'debug', 'sweet'), stdin=PIPE, stdout=PIPE)
+        proc.stdin.close()
+        try:
+            assert proc.stdout.readline() == 'hello, i am a pre-start script in stdout\n'
+        finally:
+            ctrl_c(proc)
+            proc.wait()
```

Comment:
Yep, that is another approach, but I'm not seeing how your suggestion is any better than this one. Is this a blocking concern for you?